apps/core/app/[locale]/(default)/cart/_components/cart-item.tsx

```diff
@@ -0,0 +1,122 @@
+import { Trash } from 'lucide-react';
+import { getTranslations } from 'next-intl/server';
+
+import { getCart } from '~/client/queries/get-cart';
+import { ExistingResultType } from '~/client/util';
+import { BcImage } from '~/components/bc-image';
+
+import { removeProduct } from '../_actions/remove-products';
+
+import { CartItemCounter } from './cart-item-counter';
+
+type Product =
+  | ExistingResultType<typeof getCart>['lineItems']['physicalItems'][number]
+  | ExistingResultType<typeof getCart>['lineItems']['digitalItems'][number];
+
+export const CartItem = async ({ product }: { product: Product }) => {
+  const t = await getTranslations('Cart');
+
+  const extractCartlineItemsData = ({
```

Comment:
🍹 Not from this PR but maybe a quick win, thoughts on changing `CartItemCounter` to avoid this?

Reply:
Like pass in props instead of a single object?